core/rid.h

```diff
@@ -78,7 +78,16 @@ class RID {
 	}
 	_FORCE_INLINE_ bool is_valid() const { return _data != nullptr; }
 
-	_FORCE_INLINE_ uint32_t get_id() const { return _data ? _data->get_id() : 0; }
+	_FORCE_INLINE_ uint32_t get_id() const {
+		if (_data) {
+			try {
+				return _data->get_id();
```

Comment:
Is it necessary for me to `try` here? Or are exceptions handled elsewhere? Since gdscript can still create dangling pointers, it's possible get_id() will reference invalid memory. I tried freeing an RID, then getting it's ID in gdscript, in another build of the engine. I got a non-zero number, but no crash or error message.

Reply:
Some of the mobile platform builds say `error: cannot use 'try' with exceptions disabled`. So should I not `try`?

Reply:
I see here the bigges problem with this approach.

In Godot we're not using exceptions (some 3rd-prty code may be, but Godot's own codebase isn't), so `try/catch` is forbidden. However, let's consider for a moment if the use case here is strong enough. If I'm not mistaken, what you are attempting to catch here is a segfault (i.e., accessing memory no longer owned by the process), being that a way to detect an `RID` is dangling. But wouldn't it be possible that a future `RID` data ends up being alocated at the same (or overlapping) memory address, rendering that sort of check useless?

That's a bit like the dangling `Variant` issue we use to have.

Reply:
The PR is not dependent upon try/catch. This bit is optional to prevent potential issues in gdscript, though as I said it doesn't seem to matter. The key part that fixed the dangling pointer is _free() where _data is set to nullptr.